### PR TITLE
SHARE-346 API Search users

### PR DESCRIPTION
**Pull request by JackShahhoud**
-Implement Search users
-Write tests

---
### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroweb-Symfony/blob/develop/.github/contributing.md) and [wiki pages](https://github.com/Catrobat/catroweb-Symfony/wiki/) of this repository.

- [x] Include the name and id of the Jira ticket in the PR’s title
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no warnings and errors 
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Verify that all tests are passing, if not please state the test cases in the [section](#Tests) below
- [x] Perform a self-review of the changes
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] Put your ticket into the `Code Review` section in [Jira](https://jira.catrob.at/)
- [x] Post a message in the *#catroweb* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: tests/behat/features/api/user/search_users.feature [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    Given I have a parameter "query" with value "Catroweb"
+    And I have a request header "HTTP_ACCEPT" with value "application/json"
+    And I request "GET" "/api/users/search"
+    Then the response status code should be "200"
+    Then the response should have the users model structure
+    Then the response should contain users in the following order:
+      | Name      |
+      | Catroweb  |
+
```
Comment (dmetzner) [inline tests/behat/features/api/user/search_users.feature:80]: we have many russian users, a test with russian chars would be nice :)

**Review by dmetzner — CHANGES_REQUESTED** [review]
-  maybe add the russian character testcase.

- Concerning the image... we should discuss this with the teams. But definetly we should only use one variant. Url or base64 image - but not both. @HCrane 

 If we use the url variant -> clear/small api response, but second call needed to get the image + a rework of the profile image system is necessary or at least an additional api. (Similar to github downlaod artifacts)

if we use base 64 variant -> not much to do, just instead of the default image return 
a) the default image in base 64 format
b) return an empty string
+ we slightly need to adapt the description on Catroweb-API repo

- the search itself is interesting - maybe you can explain it:
```

query = 'a' => returns [ Angela ]

query = 'catro' =>return [ catroweb ]

query = 'c' => return [ all users .. i guess because of the localhost email?]   

query = 'o' => returns nothing... so maybe not the localhost mail address...

query = 'atroweb' returns nothing .. why is the 'c' so important here?

query = 'ca' returns [ catroweb ]
```

is it possible to reduce the importance of the 1st char?

rest looks good to me

File: tests/behat/features/api/user/search_users.feature (hunk 1 above)
Comment (JackShahhoud) [reply tests/behat/features/api/user/search_users.feature:80]: Good Idea, although i don't speak Russian, i will try to write tests using Russian alphabet. @dmetzner

Comment (HCrane) [reply tests/behat/features/api/user/search_users.feature:80]: Google Translate is our friend! Dosen't have to be meaningfull names just russian chars!

**Review by dmetzner — APPROVED** [review]

**Pull request merged**